cpp/src/ghc/filesystem.hpp

```diff
@@ -0,0 +1,5526 @@
+//---------------------------------------------------------------------------------------
```

Comment:
I would suggest we name it `cpp/src/ghc-filesystem/`

Add this file to `.clang-format-hook-ignore` so that it is not reformatted by accident

Also, add a `ghc-filesystem.origin` file similar to `cpp/src/edn-cpp/edn-cpp.origin` (maybe use the release tag instead of or in addition to commit hash)

Reply:
Done.